<h3 id="jobsubmissionthroughstampede2">6. Steps for Running Additional Applications from Stampede2 or Lonestar5</h2>

<h4 id="prerequisites"><strong>Prerequisites</strong></h4>

<p>
  Instructions below should be run from Stampede2 or Lonestar5 supercomputers.
</p>

<p>
  A separate registration step is not required for users who are submitting jobs for the first time from Stampede2 or Lonestar5 systems. The registration process is automatically taken care of at the time of submitting jobs to BOINC@TACC from Stamepede2 or Lonestar5.
</p>

<h4 id="installation"><strong>Installation</strong></h4>

<p>
  Obtain the required files from <a href="https://github.com/ritua2/BOINCatTACC/tree/master/stampede2-backend/S2-test-files">this directory</a>:
</p>

<pre><code class="bash language-bash">svn checkout https://github.com/ritua2/BOINCatTACC/trunk/stampede2-backend/S2-test-files;
cd S2-test-files;
chmod +x advance-submit.sh;</code></pre>

<h4 id="accessingthetutorialfilesforspecificapplications"><strong>Tutorial files for TACC specific applications</strong></h4>

<p>
  A sample exercise for each specific application is provided below. All the necessary files are available when cloning the github repo as per the above instructions. However, for users wishing to use only one application, follow the instructions below:
</p>

<p>
  Note: The script also accounts for job submission using SLURM. In order to utilize BOINC, the job must use 5 or less cores, and below 2048 MB of RAM.
</p>

<ul>
  <li>BOINC job submission</li>

<pre><code class="bash language-bash">curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/advance-submit.sh;
chmod +x advance-submit.sh;
</code></pre>

</ul>

<ol>
  <li>AutoDock-Vina
    <ul>
      <li>Obtain the input and data files:</li>

      <pre><code class="bash language-bash">curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/protein.pdbqt;
curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/protein.pdb;
curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/ligand.pdbqt;
curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/ligand.pdb;
curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/ligand_experiment.pdbqt;
curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/conf.txt;
</code></pre>


      <li>Run job submission script:</li>

      <pre><code class="bash language-bash">./advance-submit.sh;</code></pre>

      <li><p>Provide your email when requested</p></li>

      <li><p>Select <em>1</em> for <em>allowed options</em> when prompted  </p></li>

      <li><p>Select <em>1</em> for applications when prompted</p></li>

      <li><p>Continue with the following instructions to run a TACC training file  </p>

      <ul>
        <li><p>Provide the following files (space separated):</p>

          <pre><code class="bash language-bash">protein.pdbqt ligand.pdbqt ligand.pdb protein.pdb ligand_experiment.pdb ligand_experiment.pdbqt conf.txt</code></pre>
        </li>

        <li><p>No directories must be provided (enter to skip)</p></li>

        <li>
          <p>Provide the following command (line by line, empty line at the end to exit):</p>

          <pre><code class="bash language-bash">vina --config conf.txt</code></pre>
        </li>
      </ul>
    </ul>
  </li>



  <li>Bedtools

    <ul>
      <li>Obtain the input and data files:</li>

      <pre><code class="bash language-bash">curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/cpg.bed;
curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/exons.bed;
</code></pre>


      <li>Run job submission script:</li>

      <pre><code class="bash language-bash">./advance-submit.sh;</code></pre>

      <li><p>Provide your email when requested</p></li>

      <li><p>Select <em>1</em> for <em>allowed options</em> when prompted  </p></li>

      <li><p>Select <em>2</em> for applications when prompted</p></li>

      <li><p>Continue with the following instructions to run a TACC training file  </p>

      <ul>
        <li><p>Provide the following files (space separated):</p>

          <pre><code class="bash language-bash">cpg.bed exons.bed</code></pre>
        </li>

        <li><p>No directories must be provided (enter to skip)</p></li>

        <li>
          <p>Provide the following command (line by line, empty line at the end to exit):</p>

          <pre><code class="bash language-bash">bedtools intersect -a cpg.bed -b exons.bed &gt; bed_test.txt</code></pre>
        </li>
      </ul>
    </ul>
  </li>



  <li>Blast

    <ul>
      <li>Obtain the input and data files:</li>

      <pre><code class="bash language-bash">curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/P04156.fasta;
curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/zebrafish.1.protein.faa.gz;
</code></pre>


      <li>Run job submission script:</li>

      <pre><code class="bash language-bash">./advance-submit.sh;</code></pre>

      <li><p>Provide your email when requested</p></li>

      <li><p>Select <em>1</em> for <em>allowed options</em> when prompted  </p></li>

      <li><p>Select <em>3</em> for applications when prompted</p></li>

      <li><p>Continue with the following instructions to run a TACC training file  </p>

      <ul>
        <li><p>Provide the following files (space separated):</p>

          <pre><code class="bash language-bash">P04156.fasta zebrafish.1.protein.faa.gz</code></pre>
        </li>

        <li><p>No directories must be provided (enter to skip)</p></li>

        <li>
          <p>Provide the following command (line by line, empty line at the end to exit):</p>

          <pre><code class="bash language-bash">gunzip zebrafish.1.protein.faa.gz
makeblastdb -in zebrafish.1.protein.faa -dbtype prot
blastp -query P04156.fasta -db zebrafish.1.protein.faa -out results.txt
</code></pre>
        </li>
      </ul>
    </ul>
  </li>



  <li>Bowtie

    <ul>
      <li>Obtain the input and data files:</li>

      <pre><code class="bash language-bash">curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/s_cerevisiae.ebwt.zip;</code></pre>


      <li>Run job submission script:</li>

      <pre><code class="bash language-bash">./advance-submit.sh;</code></pre>

      <li><p>Provide your email when requested</p></li>

      <li><p>Select <em>1</em> for <em>allowed options</em> when prompted  </p></li>

      <li><p>Select <em>4</em> for applications when prompted</p></li>

      <li><p>Continue with the following instructions to run a TACC training file  </p>

      <ul>
        <li>
          <p>Provide the following files (space separated):</p>

          <pre><code class="bash language-bash">s_cerevisiae.ebwt.zip</code></pre>
        </li>

        <li><p>No directories must be provided (enter to skip)</p></li>

        <li><p>Provide the following command (line by line, empty line at the end to exit):</p>

          <pre><code class="bash language-bash">unzip s_cerevisiae.ebwt.zip
bowtie -c s_cerevisiae ATTGTAGTTCGAGTAAGTAATGTGGGTTTG &gt; res_bowtie.txt
</code></pre>
        </li>
      </ul>
    </ul>
  </li>



  <li>Gromacs

    <ul>
      <li>Obtain the input and data files:</li>

      <pre><code class="bash language-bash">curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/methane_water.gro;
curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/topol.top;
curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/em_steep.mdp;</code></pre>


      <li>Run job submission script:</li>

      <pre><code class="bash language-bash">./advance-submit.sh;</code></pre>

      <li><p>Provide your email when requested</p></li>

      <li><p>Select <em>1</em> for <em>allowed options</em> when prompted  </p></li>

      <li><p>Select <em>5</em> for applications when prompted</p></li>

      <li><p>Continue with the following instructions to run a TACC training file  </p>

      <ul>
        <li>
          <p>Provide the following files (space separated):</p>

          <pre><code class="bash language-bash">methane_water.gro topol.top em_steep.mdp</code></pre>
        </li>

        <li><p>No directories must be provided (enter to skip)</p></li>

        <li><p>Provide the following command (line by line, empty line at the end to exit):</p>

          <pre><code class="bash language-bash">gmx grompp -f em_steep.mdp -c methane_water.gro -p topol.top -o min_thing.tpr</code></pre>
        </li>
      </ul>
    </ul>
  </li>



  <li>HTSeq

    <ul>
      <li>Obtain the input and data files:</li>

      <pre><code class="bash language-bash">curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/test.sam;
curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/gen.gtf;</code></pre>


      <li>Run job submission script:</li>

      <pre><code class="bash language-bash">./advance-submit.sh;</code></pre>

      <li><p>Provide your email when requested</p></li>

      <li><p>Select <em>1</em> for <em>allowed options</em> when prompted  </p></li>

      <li><p>Select <em>6</em> for applications when prompted</p></li>

      <li><p>Continue with the following instructions to run a TACC training file  </p>

      <ul>
        <li>
          <p>Provide the following files (space separated):</p>

          <pre><code class="bash language-bash">test.sam gen.gtf</code></pre>
        </li>

        <li><p>No directories must be provided (enter to skip)</p></li>

        <li><p>Provide the following command (line by line, empty line at the end to exit):</p>

          <pre><code class="bash language-bash">htseq-count test.sam gen.gtf &gt; test_htseq.txt</code></pre>
        </li>
      </ul>
    </ul>
  </li>



  <li>MPI-LAMMPS

    <ul>
      <li>Obtain the input and data files:</li>

      <pre><code class="bash language-bash">curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/Al99.eam.alloy;
curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/aluminium.in;</code></pre>


      <li>Run job submission script:</li>

      <pre><code class="bash language-bash">./advance-submit.sh;</code></pre>

      <li><p>Provide your email when requested</p></li>

      <li><p>Select <em>1</em> for <em>allowed options</em> when prompted  </p></li>

      <li><p>Select <em>7</em> for applications when prompted</p></li>

      <li><p>Continue with the following instructions to run a TACC training file  </p>

      <ul>
        <li>
          <p>Provide the following files (space separated):</p>

          <pre><code class="bash language-bash">Al99.eam.alloy aluminium.in</code></pre>
        </li>

        <li><p>No directories must be provided (enter to skip)</p></li>

        <li><p>Provide the following command (line by line, empty line at the end to exit):</p>

          <pre><code class="bash language-bash">mpirun -np 2 lmp_mpi &lt; aluminium.in &gt; lammps_test.txt</code></pre>
        </li>
      </ul>
    </ul>
  </li>



  <li>NAMD

    <ul>
      <li>Obtain the input and data files:</li>

      <pre><code class="bash language-bash">curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/alanin.psf;
curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/alanin.conf;
curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/alanin.pdb;
curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/alanin.params;</code></pre>
      
      <li>Note: If running this example using the slurm functionality, update the <em>alanin.conf</em> file, line 15, to your local path</li>

      <li>Run job submission script:</li>

      <pre><code class="bash language-bash">./advance-submit.sh;</code></pre>

      <li><p>Provide your email when requested</p></li>

      <li><p>Select <em>1</em> for <em>allowed options</em> when prompted  </p></li>

      <li><p>Select <em>8</em> for applications when prompted</p></li>

      <li><p>Continue with the following instructions to run a TACC training file  </p>

      <ul>
        <li>
          <p>Provide the following files (space separated):</p>

          <pre><code class="bash language-bash">alanin.psf alanin.conf alanin.pdb alanin.params</code></pre>
        </li>

        <li><p>No directories must be provided (enter to skip)</p></li>

        <li><p>Provide the following command (line by line, empty line at the end to exit):</p>

          <pre><code class="bash language-bash">namd2 alanin.conf &gt; results_namd.res</code></pre>
        </li>
      </ul>
    </ul>
  </li>



  <li>CUDA

    <ul>
      <li>Obtain the input and data files:</li>

      <pre><code class="bash language-bash">curl -O https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/happ.cu;</code></pre>

      <li>Run job submission script:</li>

      <pre><code class="bash language-bash">./advance-submit.sh;</code></pre>

      <li><p>Provide your email when requested</p></li>

      <li><p>Select <em>1</em> for <em>allowed options</em> when prompted  </p></li>

      <li><p>Select <em>10</em> for applications when prompted</p></li>

      <li><p>Continue with the following instructions to run a TACC training file  </p>

      <ul>
        <li>
          <p>Provide the following files (space separated):</p>

          <pre><code class="bash language-bash">happ.cu</code></pre>
        </li>

        <li><p>No directories must be provided (enter to skip)</p></li>

        <li><p>Provide the following command (line by line, empty line at the end to exit):</p>

          <pre><code class="bash language-bash">nvcc happ.cu -o h.out
./h.out</code></pre>
        </li>
      </ul>
    </ul>
  </li>



  <li>OpenFOAM6

    <ul>
      <li>Obtain the input and data files:</li>

      <pre><code class="bash language-bash">mkdir 0 constant system;
curl https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/0/U &gt; 0/U;
curl https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/0/p &gt; 0/p;
curl https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/constant/transportProperties &gt; constant/transportProperties;
mkdir constant/polyMesh;
curl https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/constant/polyMesh/boundary  &gt; constant/polyMesh/boundary;
curl https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/constant/polyMesh/faces     &gt; constant/polyMesh/faces;
curl https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/constant/polyMesh/neighbour &gt; constant/polyMesh/neighbour;
curl https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/constant/polyMesh/owner     &gt; constant/polyMesh/owner;
curl https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/constant/polyMesh/points    &gt; constant/polyMesh/points;
curl https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/system/blockMeshDict        &gt; system/blockMeshDict;
curl https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/system/controlDict          &gt; system/controlDict;
curl https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/system/fvSchemes            &gt; system/fvSchemes;
curl https://raw.githubusercontent.com/ritua2/BOINCatTACC/master/stampede2-backend/S2-test-files/system/fvSolution           &gt; system/fvSolution;</code></pre>

      <li>Run job submission script:</li>

      <pre><code class="bash language-bash">./advance-submit.sh;</code></pre>

      <li><p>Provide your email when requested</p></li>

      <li><p>Select <em>1</em> for <em>allowed options</em> when prompted  </p></li>

      <li><p>Select <em>11</em> for applications when prompted</p></li>

      <li><p>Continue with the following instructions to run a TACC training file  </p>

      <ul>
        <li><p>No input files are needed (enter to skip):</p></li>

      <li>Provide the following directories, each in a new line:
        <pre><code class="bash language-bash">0
constant
system</code></pre>
      </li>

        <li><p>Provide the following command (line by line, empty line at the end to exit):</p>

          <pre><code class="bash language-bash">blockMesh
icoFoam</code></pre>
        </li>
      </ul>
    </ul>
  </li>
</ol>
